src/main/java/edu/cornell/kfs/fp/service/impl/RecurringDisbursementVoucherDocumentServiceImpl.java

```diff
@@ -409,8 +417,117 @@ public Object call() throws WorkflowException {
+                    if (isKfsSystemUserDvInitiator(dv)) {
+                        try {
+                            addNoteToAutoApproveDv(dv, "Batch job Submit and Blanket Approve performed for DV spawned by Recurring DV.");
+                            try{
+                                blanketApproveDisbursementVoucherDocument(dv);
+                                approvalCount++;
+
+                            } catch (WorkflowException e) {
+                                errorCount++;
+                            }
+                        } catch (WorkflowException e) {
+                            errorCount++;
+                        }
+                    }
+                    else {
+                        LOG.error("Detected document initiator was not KFS System user. Auto blanket approval was NOT attemtped for document ID: " + dv.getDocumentNumber());
+                        errorCount++;
+                    }
+                }
+            }
+        }
+        else {
+            LOG.info("No DV's spwned by Recurring DV found. Nothing will be auto approved. ");
+        }
+        LOG.info("Number of Disbursement Vouchers that generated Errors:: " + errorCount);
+        LOG.info("Number of Disbursement Vouchers successfuly Blanket Approved (fully processed):: " + approvalCount);
+        LOG.info("autoApproveDisbursementVouchersSpawnedByRecurringDvs: Leaving.");
+        return errorCount == 0;
+    }
     
-    
+    private boolean isKfsSystemUserDvInitiator(CuDisbursementVoucherDocument dv) {
+        String dvDocInitatorPrincipalId = dv.getDocumentHeader().getWorkflowDocument().getInitiatorPrincipalId();
+
+        Person kfsSystemUser = getPersonService().getPersonByPrincipalName(KFSConstants.SYSTEM_USER);
+        String kfsSystemUserPrincipalId = kfsSystemUser.getPrincipalId();
+
+        if ( (dvDocInitatorPrincipalId != null) && (kfsSystemUserPrincipalId != null) &&
+             (!StringUtils.isBlank(dvDocInitatorPrincipalId)) && (!StringUtils.isBlank(kfsSystemUserPrincipalId)) &&
+             (StringUtils.equals(dvDocInitatorPrincipalId, kfsSystemUserPrincipalId)) )
+            return true;
+        else
+            return false;
+    }
+
+    private void addNoteToAutoApproveDv(CuDisbursementVoucherDocument dv, String noteText) throws WorkflowException {
+        Note note = buildNoteBase();
+        note.setNoteText(noteText);
+        dv.addNote(note);
+        try {
+            getDocumentService().saveDocument(dv);
+        } catch (WorkflowException e) {
+            LOG.error("addNoteToAutoApproveDv: Unable to save note for DV Document ID::" + dv.getDocumentNumber() + "the exception was: ", e);
+            throw e;
+        }
+    }
+
+    private void blanketApproveDisbursementVoucherDocument(CuDisbursementVoucherDocument dv) throws WorkflowException {
+        try {
+            getDocumentService().blanketApproveDocument(dv, "Auto blanket approve from Batch Job", null);
```

Comment:
not sure it makes sense to have this call in a separate method since that is all it does